src/cbc_sdk/enterprise_edr/base.py

```diff
@@ -272,38 +270,189 @@ def __init__(self, doc_class, cb):
         super(Query, self).__init__(doc_class, cb, None)
 
         self._query_builder = QueryBuilder()
-        self._sort_by = None
+        self._criteria = {}
+        self._exclusions = {}
+        self._sort_by = []
         self._group_by = None
-        self._batch_size = 100
+        self._batch_size = 500
+        self._start = 0
+        self._time_range = {}
+        self._fields = ["*"]
         self._default_args = {}
 
+    def update_criteria(self, key, newlist):
+        """Update the criteria on this query with a custom criteria key.
+
+        Args:
+            key (str): The key for the criteria item to be set.
+            newlist (list): List of values to be set for the criteria item.
+
+        Returns:
+            The ResultQuery with specified custom criteria.
+
+        Example:
+            query = api.select(Event).update_criteria("my.criteria.key", ["criteria_value"])
+        """
+        if not isinstance(newlist, list):
+            raise ApiError(f"Criteria value(s) must be a list. {newlist} is a {type(newlist)}.")
+        self._update_criteria(key, newlist)
+        return self
+
+    def _update_criteria(self, key, newlist):
+        """
+        Updates a list of criteria being collected for a query, by setting or appending items.
+
+        Args:
+            key (str): The key for the criteria item to be set.
+            newlist (list): List of values to be set for the criteria item.
+        """
+        oldlist = self._criteria.get(key, [])
+        self._criteria[key] = oldlist + newlist
+
+    def update_exclusions(self, key, newlist):
+        """Update the criteria on this query with a custom exclusion key.
+
+        Args:
+            key (str): The key for the exclusion item to be set.
+            newlist (list): List of values to be set for the exclusion item.
+
+        Returns:
+            The ResultQuery with specified custom exclusion.
+
+        Example:
+            query = api.select(Event).update_exclusions("my.exclusion.key", ["exclusion_value"])
+        """
+        if not isinstance(newlist, list):
+            raise ApiError(f"Exclusion value(s) must be a list. {newlist} is a {type(newlist)}.")
+        self._update_exclusions(key, newlist)
+        return self
+
+    def _update_exclusions(self, key, newlist):
+        """
+        Updates a list of exclusion being collected for a query, by setting or appending items.
+
+        Args:
+            key (str): The key for the exclusion item to be set.
+            newlist (list): List of values to be set for the exclusion item.
+        """
+        oldlist = self._exclusions.get(key, [])
+        self._exclusions[key] = oldlist + newlist
+
+    def update_fields(self, fields):
```

Comment:
It seems odd to have these (`update_fields`, `update_start`, `update_rows`, `update_time_range`) called update as they are more setters unlike criteria and exclusions where we are adding additional key/values to the request.

I would either go with `cb.select(Process).set_rows()` or just `cb.select(Process).rows()`

Now that I am thinking about this would it make sense to make `update_criteria` --> `add_criteria`?

Reply:
This would also match closer to `sort_by` function naming

Reply:
I think the methods for time_range, fields, start, and rows would make more sense as `set_*()` or `add_*()` since they're adding or overwriting the existing values. The criteria and exclusions methods are adding or updating the existing values, so those make more sense as `update_*()` or `add_*()`.

Reply:
I wouldn't do `add_rows` as that reads a little strange `set_rows` makes more sense. Then I would go with `add_criteria` as update would make more sense if we were updating an existing criteria property like device_ids that is already set